server/app/services/applicant/ApplicantData.java

```diff
@@ -150,4 +152,15 @@ public ImmutableMap<Path, String> getFailedUpdates() {
   public boolean updateDidFailAt(Path path) {
     return getFailedUpdates().containsKey(path);
   }
+
+  public Optional<LocalDate> getDateOfBirth() {
+    return readDate(WellKnownPaths.APPLICANT_DOB);
+  }
+
+  public void setDateOfBirth(String dateString) throws DateOfBirthExistsException {
+    if (hasPath(WellKnownPaths.APPLICANT_DOB)) {
+      throw new DateOfBirthExistsException();
+    }
```

Comment:
question (blocking): why can the date of birth not change after initially set? what if a user needs to correct a mistake?